docs/specifications/tests/Connectivity-TP/connectivity02.md

```diff
@@ -1,48 +1,194 @@
+            *[CN02_UNEXPECTED_RCODE_SOA_QUERY_TCP]* with [RCODE Name] and name
+            and IP address of the name server.
+         3. Else, if there is no SOA record in the answer section, then
+            output *[CN02_MISSING_SOA_RECORD_TCP]* with name and IP address of
+            the name server.
+         4. Else, if the SOA record has owner name other than *Child Zone*
+            then output *[CN02_WRONG_SOA_RECORD_TCP]* with name and IP address of
+            the name server, the SOA record owner name and *Child Zone*.
+         5. Else, if AA flag is unset, then output *[CN02_SOA_RECORD_NOT_AA_TCP]*
+            with name and IP address of the name server.
+      2. Process the response on *NS Query TCP*:
+         1. If there is no [DNS Response], then output
+            *[CN02_NO_RESPONSE_NS_QUERY_TCP]* with name and IP address of the
+            name server.
+         2. Else, if [RCODE Name] is not "NoError" then output
+            *[CN02_UNEXPECTED_RCODE_NS_QUERY_TCP]* with [RCODE Name] and name and
+            IP address of the name server.
+         3. Else, if there is no NS record in the answer section, then
+            output *[CN02_MISSING_NS_RECORD_TCP]* with name and IP address of the
+            name server.
+         4. Else, if the NS record has owner name other than *Child Zone*
+            then output *[CN02_WRONG_NS_RECORD_TCP]* with name and IP address of
+            the name server, the NS record owner name and *Child Zone*.
+         5. Else, if AA flag is unset, then output *[CN02_NS_RECORD_NOT_AA_TCP]*
+            with name and IP address of the name server.
 
-1. Obtains the IP addresses of the name servers from [Method4] and
-   [Method5].
-2. A SOA query is sent over TCP to distinct IP address of each name server
-   found in step 1.
-3. If all queries in step 2 receive a DNS answer (bogus response are not
-   checked here) then the test case succeed.
 
 ## Outcome(s)
 
-If there is any name server that fails to answer queries over port 53 using
-TCP, this test case fails.
+The outcome of this Test Case is "fail" if there is at least one message
+with the severity level *[ERROR]* or *[CRITICAL]*.
 
-## Special procedural requirements
+The outcome of this Test Case is "warning" if there is at least one message
+with the severity level *[WARNING]*, but no message with severity level
+*ERROR* or *CRITICAL*.
 
-If either IPv4 or IPv6 transport is disabled, ignore the evaluation of the
-result of any test using this transport protocol. Log a message reporting
-on the ignored result.
+In other cases, no message or only messages with severity level
+*[INFO]* or *[NOTICE]*, the outcome of this Test Case is "pass".
 
-## Intercase dependencies
 
-None
+## Special procedural requirements
+
+If either IPv4 or IPv6 transport is disabled, skip sending queries over that
+transport protocol. A message will be outputted reporting that the transport
+protocol has been skipped.
```

Comment:
This can be removed I think, this Test Case will not output anything for disabled IPvX now.

Reply:
I assume you only mean the last sentence. -- This text is taken in verbatim from the template, and the same text or a variant of it is found in all or many test case specifications. -- Log messages are outputted, but on debug level.

For the specification I do not think the sentence is meaningful. I remove it, and create an update for the template.

Reply:
Well new Connectivity01 specification will output NOTICE level messages regarding IPvX transport (for now at least). Instead maybe its better if its explicitly specified in the list in the "Objective" section of the specification of Connectivity01? [Link](https://github.com/zonemaster/zonemaster/blob/89ea9e1ebac0e9ab1dd99ecf2c481634730e5c77/docs/specifications/tests/Connectivity-TP/connectivity01.md?plain=1#L36-L44)

Reply:
Something like:

`* Disabled IP transport protocol, and the list of the affected Name Servers`

Reply:
But not in this specification, I assume.

Reply:
Yes, that is what I meant, explicitly specify it only in Connectivity01.